### PR TITLE
If softdoom factor has been overridden, we shall not adjust the autom…

### PR DESCRIPTION
**Pull request by baldersheim**
…atic one.

@havardpe PR

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/matching/matcher.cpp [hunk 1]
```diff
@@ -286,19 +291,20 @@ Matcher::match(const SearchRequest &request, vespalib::ThreadBundle &threadBundl
...
-            fastos::TimeStamp softLimit = (1.0 - _rankSetup->getSoftTimeoutTailCost()) * request.getTimeout();
+            fastos::TimeStamp overtimeLimit = (1.0 - _rankSetup->getSoftTimeoutTailCost()) * request.getTimeout();
             fastos::TimeStamp adjustedDuration = duration - my_stats.doomOvertime();
             if (adjustedDuration < 0) {
                 adjustedDuration = 0;
             }
             bool allowedSoftTimeoutFactorAdjustment = (std::chrono::duration_cast<std::chrono::seconds>(my_clock::now() - _startTime).count() > SECONDS_BEFORE_ALLOWING_SOFT_TIMEOUT_FACTOR_ADJUSTMENT);
-            if (allowedSoftTimeoutFactorAdjustment) {
-                _stats.updatesoftDoomFactor(request.getTimeout(), softLimit, adjustedDuration);
+            if (allowedSoftTimeoutFactorAdjustment && ! isDoomExplicit) {
```
Comment (havardpe) [inline searchcore/src/vespa/searchcore/proton/matching/matcher.cpp]: consider making the explicit doom part of whether we are allowed to tamper, not an additional reason not to

**Review by havardpe — APPROVED** [review]
maybe the explicitness of the soft doom should be modeled as part of Doom...?

File: searchcore/src/vespa/searchcore/proton/matching/matcher.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/matching/matcher.cpp]: Done

**Comment by baldersheim** [comment]
Superceeded by #11479

**Pull request closed without merging**